### PR TITLE
Port #51205 to master

### PR DESCRIPTION
**Pull request by terminalmage**
See https://github.com/saltstack/salt/pull/51205

```
Lint: Fix str-format-in-logging, len-as-condition, etc. (6)

This fixes violations found in the following locations:

- salt/state.py
- salt/states/
- salt/tops/
- salt/transport/
```

### CODE REVIEW

**Review by waynew — CHANGES_REQUESTED** [review]
Thanks for the port 🚀 

Just one tweak (-six) and one question (assert??) and should be good 👍

**Comment by waynew** [comment]
I'm guessing the test failures are fixed on master, too, FWIW. At a glance they don't look related.

File: salt/states/btrfs.py [hunk 1]
```diff
@@ -104,8 +103,8 @@ def wrapper(*args, **kwargs):
                 kwargs["__dest"] = dest
             ret = action(*args, **kwargs)
         except Exception as e:  # pylint: disable=broad-except
-            log.error("""Traceback: {}""".format(traceback.format_exc()))
-            ret["comment"].append(e)
+            log.exception("Error encountered running %s", action.__name__)
+            ret["comment"].append(six.text_type(e))
```
[the remark this answers is not in the document]
Comment (terminalmage) [reply salt/states/btrfs.py:107]: Done.

File: salt/states/ldap.py [hunk 2]
```diff
@@ -441,7 +441,7 @@ def _process_entries(l, entries):
                     )
                 else:
                     # nothing, so it must be a brand new entry
-                    assert len(results) == 0
+                    assert not results
```
[the remark this answers is not in the document]
Comment (terminalmage) [reply salt/states/ldap.py:444]: Actually, I'm not going to make any changes for this one. There are in total three asserts in this module, and while to be honest we should never use them in non-test code, both of the following are true:

1. I did not write this code
2. I do not understand how it is supposed to work.

In the interest of not introducing new bugs, I think that a new issue should be opened to replace all the asserts in this module.

At first glance, I don't think any of these asserts should be fatal for the state. It's possible that they were once used as a crude signaling measure, but there no longer appears to be anything catching them. But again, I'm not familiar with this code. This PR is solely to resolve lint issues.

[the remark this answers is not in the document]
Comment (waynew) [reply salt/states/ldap.py:444]: I agree with that - I wasn't sure to bring it up even, but figured I'd err on the side of calling out the oddity. If you haven't opened an issue, let me know and I can go ahead and do that.

**Review by waynew — DISMISSED** [review]

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**